index.js

```diff
@@ -221,51 +221,66 @@ const formatResults = (
 
 // Processes records and either extracts Typed Values into an array, or
 // object with named column labels
-const formatRecords = (recs,columns) => {
+const formatRecords = (recs, columns) => {
+  // Map over all the records (rows)
+  return recs
+    ? recs.map((rec) => {
+      // Reduce each field in the record (row)
 
-  // Create map for efficient value parsing
-  let fmap = recs && recs[0] ? recs[0].map((x,i) => {
-    return Object.assign({},
-      columns ? { label: columns[i].label } : {} ) // add column labels
-  }) : {}
+      return columns ? formatHydrate(rec, columns) : formatDefault(rec) // object if hydrate, else array
+    })
+    : [] // empty record set returns an array
+} // end formatRecords
 
-  // Map over all the records (rows)
-  return recs ? recs.map(rec => {
+// formats record as an object
+const formatHydrate= (rec, column) => {
+  let formattedRec = {}
 
-    // Reduce each field in the record (row)
-    return rec.reduce((acc,field,i) => {
+  rec.forEach((field, i) => {
+    // If the field is null, always return null
+    if (field.isNull === true) {
+      formattedRec[column[i].label] = null
+    } else {
+      formattedRec[column[i].label] = formatNonNullValue(field)
+    }
+  })
 
-      // If the field is null, always return null
-      if (field.isNull === true) {
-        return columns ? // object if hydrate, else array
-          Object.assign(acc,{ [fmap[i].label]: null })
-          : acc.concat(null)
+  return formattedRec
+}
 
-      // If the field is mapped, return the mapped field
-      } else if (fmap[i] && fmap[i].field) {
-        return columns ? // object if hydrate, else array
-          Object.assign(acc,{ [fmap[i].label]: field[fmap[i].field] })
-          : acc.concat(field[fmap[i].field])
+// formats record as an array
+const formatDefault = (rec) => {
+  let formattedRec = []
 
-      // Else discover the field type
-      } else {
+  rec.forEach((field) => {
+    // If the field is null, always return null
+    if (field.isNull === true) {
+      formattedRec.push(null)
+    } else {
+      formattedRec.push(formatNonNullValue(field))
+    }
+  })
 
-        // Look for non-null fields
-        Object.keys(field).map(type => {
-          if (type !== 'isNull' && field[type] !== null) {
-            fmap[i]['field'] = type
-          }
-        })
-
-        // Return the mapped field (this should NEVER be null)
-        return columns ? // object if hydrate, else array
-          Object.assign(acc,{ [fmap[i].label]: field[fmap[i].field] })
-          : acc.concat(field[fmap[i].field])
-      }
+  return formattedRec
+}
 
-    }, columns ? {} : []) // init object if hydrate, else init array
-  }) : [] // empty record set returns an array
-} // end formatRecords
+// returns the value for primitives and handles Array types
+const formatNonNullValue = (value) => {
+  const actualValue = Object.values(value)[0] // value always has only one element, the value of which is what's needed
```

Comment:
This is an assumption I made based on my experience while working on this, but I'm definitely not 100% on this. 

From what I could gather the current implementation makes the same assumption in a roundabout way of actually finding the type of the value and selecting based on that. 

I would like some confirmation that this is indeed the case.